app/services/store.js

```diff
@@ -68,6 +69,20 @@ export default DS.Store.extend({
       this.push(this.normalize('commit', commit));
     }
 
+    if (name === 'branch') {
+      // force reload of repo branches
+      // delay to resolve race between github-sync and live
+      Ember.run.later(() => {
+        const repo = this.recordForId('repo', data.repository_id);
+        repo.notifyPropertyChange('branches');
+        repo.get('branches').then(branches => {
+          this.set('branches.amount', branches.content.length);
```

Comment:
If this is used only to subscribe to in the component isn't it enough to subscribe to the repo's `branches` property (as you do `notifyPropertyChange`)?